engine/src/main/java/org/pentaho/di/pan/PanCommandExecutor.java

```diff
@@ -326,10 +327,18 @@ public Trans loadTransFromRepository( Repository repository, String dirName, Str
     return trans; // return transformation loaded from the repo
   }
 
-  public Trans loadTransFromFilesystem( final String initialDir, final String filename, final String jarFilename ) throws Exception {
+  public Trans loadTransFromFilesystem( String initialDir, String filename, String jarFilename, String base64Zip )
+    throws Exception {
 
     Trans trans = null;
 
+    if ( !Utils.isEmpty( base64Zip ) ) {
+      //expected form of filename "*.ktr"
+      String zipPath = Const.getUserHomeDirectory() + File.separator + java.util.UUID.randomUUID().toString() + ".zip";
+      filename = "zip:file:" + File.separator + zipPath + "!" + filename;
+      decodeBase64StringToFile( base64Zip, zipPath ).deleteOnExit();
```

Comment:
In the eventuality the returned `File` object is null: NPE on `.deleteOnExit()`

Reply:
so I will check != null?

Reply:
and if null throw an exception?

Reply:
I think Pan will already complain if it cannot find the file 🤔 but I could provide more information as to where? is the error log enough?

Reply:
I will throw an exception in the method so it will never get to this point with a null file